openqa-investigate

```diff
@@ -1,391 +1,178 @@
+    except subprocess.TimeoutExpired:
+        warn(f"grep was killed, possibly timed out: cmd=>grep -qPzo '{search_term}' '{log_file}'")
+        return False
+    except subprocess.CalledProcessError as e:
+        warn(f"grep failed: cmd=>grep -qPzo '{search_term}' '{log_file}' output='{e.stderr}'")
+        return False
+
+def label_on_issue(host_url, job_id, search_term, label, restart=False, force_result=None):
+    log_file = get_log_file(job_id)
+    if search_log(log_file, search_term):
+        comment_on_job(host_url, job_id, label, force_result)
+        if restart:
+            response = requests.post(f"{host_url}/jobs/{job_id}/restart")
+            response.raise_for_status()
+
+def handle_unreviewed(testurl, log_file, reason, group_id, email_unreviewed, from_email, notification_address=None, job_data=None, dry_run=False):
+    # This function has been simplified due to email sending being specific to a particular environment
+    header = f"[{testurl}]({testurl}): Unknown test issue, to be reviewed\n-> [autoinst-log.txt]({testurl}/file/autoinst-log.txt)\n"
+    excerpt = "Last lines before SUT shutdown:\n\n" + get_log_excerpt(log_file)
+
+    if email_unreviewed and group_id != 'null':
+        group_data = openqa_api_get(host_url, f"job_groups/{group_id}")
+        group_description = group_data[0].get('description', '')
+        group_mailto = re.search(r".*MAILTO: (\S+).*", group_description)
+        group_mailto = group_mailto.group(1) if group_mailto else notification_address
+        clone_id = job_data['job'].get('clone_id')
+
+        if group_mailto and not clone_id:
+            group_name = group_data[0].get('name')
+            job_name = job_data['job'].get('name')
+            job_result = job_data['job'].get('result')
+            info = f"* Name: {job_name}\n* Result: {job_result}\n* Reason: {reason}\n\nIt might be a product bug, an outdated needle, test code needing adaptation or a test infrastructure related problem.\nAdding a [bugref](http://open.qa/docs/#_bug_references) that can be [carried over](http://open.qa/docs/#carry-over) will prevent these mails for this issue. If the carry-over is not sufficient, you may want to create a ticket with [auto-review-regex](https://github.com/os-autoinst/scripts/blob/master/README.md#auto-review---automatically-detect-known-issues-in-openqa-jobs-label-openqa-jobs-with-ticket-references-and-optionally-retrigger)."
+            email = f"[{job_name}]({testurl})\n{header}\n{info}\n\n{excerpt}"
+            subject = f"Unreviewed issue (Group {group_id} {group_name})"
+            if not dry_run:
+                send_email(group_mailto, email, from_email, subject)
+
+def send_email(mailto, email, from_email, subject):
+    # This function will just print the email to the console in this example
+    print(f"To: {mailto}\nFrom: {from_email}\nSubject: {subject}\n\n{email}")
+
+def get_log_file(job_id):
+    # Placeholder for actual log file retrieval logic
```

Comment:
U can use the python doc format here. In theory they can be extracted later for documentation.